src/models/MatrixAction.ts

```diff
@@ -179,7 +179,14 @@ export class MatrixAction {
                     fileSize = " (" + Math.round(event.content.info.size / 1024) + "KB)";
                 }
 
-                const url = ContentRepo.getHttpUriForMxc(mediaUrl, event.content.url);
+                let url = ContentRepo.getHttpUriForMxc(mediaUrl, event.content.url);
+                if (forceFilename) {
+                    url += `/${forceFilename}`;
+                }
+                else if (event.content.body?.match(/\.[\w\d]{2,4}$/)) {
```

Comment:
`RegExp.test()` is slightly faster and less confusing, if you're not interested in the match.
```suggestion
                else if (/\.[\w\d]{2,4}$/.test(event.content.body)) {
```